### PR TITLE
RGLC draft change alignment

### PR DESCRIPTION
**Pull request by chris-wood**
I *think* this catches everything, and should be the last update for the document. To the folks here: please check your implementation against the main branch of HPKE ([and pending changes here](https://github.com/cfrg/draft-irtf-cfrg-hpke/pull/164)) and shout if you see a discrepancy! 🙏 

cc @dmcardle, @kevinjacobs, @rozbb, @kjacobs-moz, @blipp, @raphaelrobert, @fredericjacobs

### CODE REVIEW

**Comment by dmcardle** [comment]
Looks good. New test vectors pass for BoringSSL's C and BoGo implementations. Note that we only implement the SetupBase* and SetupPSK* functions, and only support DHKEM(X25519, HKDF-SHA256).

**Comment by kjacobs-moz** [comment]
I can confirm: the new vectors pass in NSS as well ([`SetupBase*`, `SetupPSK*`] with `DHKEM(X25519, HKDF-SHA256)` and [`ChaCha20Poly1305`, `AES-128-GCM`]. Thanks!

File: hpke_test.go [hunk 1]
```diff
@@ -543,7 +543,7 @@ func verifyParameters(tv testVector, ctx context) {
 	assertBytesEqual(tv.t, tv.suite, "Incorrect parameter 'key_schedule_context'", tv.keyScheduleContext, ctx.contextParams.keyScheduleContext)
 	assertBytesEqual(tv.t, tv.suite, "Incorrect parameter 'secret'", tv.secret, ctx.contextParams.secret)
 	assertBytesEqual(tv.t, tv.suite, "Incorrect parameter 'key'", tv.key, ctx.Key)
-	assertBytesEqual(tv.t, tv.suite, "Incorrect parameter 'nonce'", tv.nonce, ctx.Nonce)
+	assertBytesEqual(tv.t, tv.suite, "Incorrect parameter 'base_iv'", tv.baseNonce, ctx.BaseNonce)
```
Comment (rozbb) [inline hpke_test.go:546]: *base_nonce

**Comment by rozbb** [comment]
Does not validate against `rust-hpke`. The newest changes to the spec remove the space after the RFC label, i.e., it's now `HPKE-06` with no trailing space.

Lines that need changing are [here](https://github.com/cisco/go-hpke/blob/master/crypto.go#L859) and [here](https://github.com/cisco/go-hpke/blob/master/crypto.go#L872).

**Comment by chris-wood** [comment]
> Does not validate against rust-hpke. The newest changes to the spec remove the space after the RFC label, i.e., it's now HPKE-06 with no trailing space.

Great catch! Fixed. @dmcardle, @kjacobs-moz, can you please try again?

**Comment by dmcardle** [comment]
Yikes, I should have caught that dropped space.

BoringSSL's Go and C implementations are passing again after (1) dropping the space from "HPKE-06 " and (2) updating to use the "base_nonce" label.

**Comment by kjacobs-moz** [comment]
Good catch indeed. All passing here with the updates.

**Comment by rozbb** [comment]
Passes with `rust-hpke`, which supports everything except for X448, P-384, and P-521

File: crypto.go [hunk 2]
```diff
@@ -856,7 +856,7 @@ func (s hkdfScheme) Expand(prk, info []byte, outLen int) []byte {
 }
 
 func (s hkdfScheme) LabeledExtract(salt []byte, suiteID []byte, label string, ikm []byte) []byte {
-	labeledIKM := append([]byte(rfcLabel+" "), suiteID...)
+	labeledIKM := append([]byte(rfcLabel), suiteID...)
```
Comment (bifurcation) [inline crypto.go:859]: Can you just add the space to the label?

Comment (bifurcation) [reply crypto.go:859]: If the space isn't always there, that seems like a bug in the spec.

**Review by bifurcation — APPROVED** [review]
Approved with nits.

File: crypto.go (hunk 2 above)
Comment (chris-wood) [reply crypto.go:859]: We removed the space from the spec (it was previously included in the spec), so everything's aligned now.

**Pull request merged**